### PR TITLE
Codenvy 694 Fix limits checking user on ws startup

### PR DESCRIPTION
**Pull request by mshaposhnik**
Fix limits checking user on ws startup

### CODE REVIEW

**Comment by mshaposhnik** [comment]
@skabashnyuk @evoevodin

**Comment by benoitf** [comment]
if you could use a more descriptive text for the PR name

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 1]
```diff
@@ -111,9 +115,15 @@ public WorkspaceImpl startWorkspace(String workspaceId,
                                                                            ServerException,
                                                                            ConflictException {
         final WorkspaceImpl workspace = getWorkspace(workspaceId);
+        String userId;
+        try {
+            userId = userManager.getByName(workspace.getNamespace()).getId();
+        } catch (NotFoundException e) {
+            userId = getCurrentUserId();
```
Comment (evoevodin) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: Workspace start behaviour must be strict and always predictable(the same in this case), i think that you should prohibit workspace start if there is no user owning the workspace found @sleshchenko, @skabashnyuk can you please take a look

**Comment by sleshchenko** [comment]
-1 for now

As far as I understand We also should fix fetching of workspaces here
https://github.com/codenvy/codenvy/pull/319/files/5a6ff77ba5a1ae64b4b6d9c0dcefc8d629ac0487#diff-8b6273ced2f7db76f99f8877a8ae652cL177
getWorkspaces returns all workspaces which are available for given user (in fact read action)
but we should fetch all workspaces by namespace of workspace that will start

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 2]
```diff
@@ -127,8 +138,8 @@ public WorkspaceImpl startWorkspace(WorkspaceConfig config,
         checkMaxEnvironmentRam(config);
         return checkRamAndPropagateStart(config,
                                          config.getDefaultEnv(),
-                                         getCurrentUserId(),
-                                         () -> super.startWorkspace(config, namespace, isTemporary, accountId));
+                                         EnvironmentContext.getCurrent().getSubject().getUserName(),
```
Comment (sleshchenko) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: should we put namespace here?

Comment (mshaposhnik) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: Yes seems we can

Comment (sleshchenko) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: If yes, seems that tests don't cover this usecase. Can you cover this?

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java [hunk 3]
```diff
@@ -111,6 +114,14 @@ public WorkspaceImpl startWorkspace(String workspaceId,
                                                                            ServerException,
                                                                            ConflictException {
         final WorkspaceImpl workspace = getWorkspace(workspaceId);
+        try {
+            userManager.getByName(workspace.getNamespace());
+        } catch (NotFoundException e) {
+            throw new ServerException(String.format(
```
Comment (sleshchenko) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java]: Are you sure that we need this check? then why we don't need it here https://github.com/codenvy/codenvy/pull/319/files#diff-8b6273ced2f7db76f99f8877a8ae652cR138

**Comment by sleshchenko** [comment]
OK for me

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**